packages/react/src/pages/user-profile/user-profile.tsx

```diff
@@ -0,0 +1,285 @@
+                stylingMode='contained'
+                icon={isXSmall ? void 0 : 'lock'}
+                onClick={handleChangePasswordClick}
+              />
+            </div>
+          </div>
+        </ProfileCard>
+
+        <ProfileCard
+          wrapperCssClass='profile-card contacts-card'
+          title='Contacts'
+          cardData={profileData}
+          items={contactItems}
+          onDataChanged={handleDataChanged}
+        >
+          <div className='profile-card-top-item'>
+            <div className='image-wrapper'>
+              <img alt='' src='icons/at.svg' />
+            </div>
+            <div>
+              <div className='title-text'>
+                {formatPhone(profileData?.phone)}
+              </div>
+              <div className='subtitle-text with-clipboard-copy'>
+                {profileData?.email}
+                <Button
+                  icon='copy'
+                  className='copy-clipboard-button'
+                  onClick={copyToClipboard(profileData?.email)}
+                  activeStateEnabled={false}
+                  focusStateEnabled={false}
+                  hoverStateEnabled={false}
+                />
+              </div>
+            </div>
+          </div>
+        </ProfileCard>
+
+        <ProfileCard
+          wrapperCssClass='profile-card address-card'
+          title='Address'
+          cardData={profileData}
+          items={addressItems}
+          onDataChanged={handleDataChanged}
+        >
+          <div className='profile-card-top-item'>
+            <div className='image-wrapper'>
+              <img alt='' src='icons/geo-position.svg' />
+            </div>
+            <div>
+              <div className='title-text'>
+                {profileData?.address}, {profileData?.city}, {profileData?.state}, {profileData?.country}
+              </div>
+            </div>
+          </div>
+        </ProfileCard>
+      </div>
+    </ScrollView>
+  );
+};
+
+const UserProfileContentWithLoadPanel = withLoadPanel(UserProfileContent);
+
+export const UserProfile = () => {
+  const [profileData, setProfileData] = useState<Record<string, string>>();
+  const [savedProfileData, setSavedProfileData] = useState();
+  const [isLoading, setIsLoading] = useState(true);
+  const [isChangePasswordPopupOpened, setIsChangedPasswordPopupOpened] = useState(false);
+  const [isDataChanged, setIsDataChanged] = useState(false);
+  const [basicInfoItems, setBasicInfoItems] = useState<ProfileCardItem[]>([]);
+  const [contactItems, setContactItems] = useState<ProfileCardItem[]>([]);
+  const [addressItems, setAddressItems] = useState<ProfileCardItem[]>([]);
+  const [isContentScrolled, setIsContentScrolled] = useState(false);
+
+  const dataChanged = useCallback(() => {
+    setIsDataChanged(true);
+  }, []);
+
+  const changePassword = useCallback(() => {
+    setIsChangedPasswordPopupOpened(true);
+  }, []);
+
+  const handleContentScrolled = (reachedTop) => {
```

Comment:
It might be useful to wrap handleContentScrolled in useCallback to reduce UserProfileContentWithLoadPanel rerenders
And setSavedData too, also add setSavedData in dependency array to onCancel and onSave (even if it is not changing now this can be useful later)